internal/riverinternaltest/riverdrivertest/riverdrivertest.go

```diff
@@ -1646,6 +1729,256 @@ func ExerciseExecutorFull[TTx any](ctx context.Context, t *testing.T, driver riv
+			require.ErrorIs(t, err, rivertype.ErrNotFound)
+			require.Nil(t, queueFetched)
+		})
+
+		t.Run("QueueList", func(t *testing.T) {
+			t.Parallel()
+
+			exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+			requireQueuesEqual := func(t *testing.T, target, actual *rivertype.Queue) {
+				t.Helper()
+				require.WithinDuration(t, target.CreatedAt, actual.CreatedAt, time.Millisecond)
+				require.Equal(t, target.Metadata, actual.Metadata)
+				require.Equal(t, target.Name, actual.Name)
+				if target.PausedAt == nil {
+					require.Nil(t, actual.PausedAt)
+				} else {
+					require.NotNil(t, actual.PausedAt)
+					require.WithinDuration(t, *target.PausedAt, *actual.PausedAt, time.Millisecond)
+				}
+			}
+
+			queues, err := exec.QueueList(ctx, 10, 0)
+			require.NoError(t, err)
+			require.Empty(t, queues)
+
+			// Make queue1, pause it, refetch:
+			queue1 := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Metadata: []byte(`{"foo": "bar"}`), Name: ptrutil.Ptr("queue1")})
+			require.NoError(t, exec.QueuePause(ctx, queue1.Name))
+			queue1, err = exec.QueueGet(ctx, queue1.Name)
+			require.NoError(t, err)
+
+			queue2 := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Name: ptrutil.Ptr("queue2")})
+			queue3 := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Name: ptrutil.Ptr("queue3")})
+
+			queues, err = exec.QueueList(ctx, 2, 0)
+			require.NoError(t, err)
+
+			require.Len(t, queues, 2)
+			requireQueuesEqual(t, queue1, queues[0])
+			requireQueuesEqual(t, queue2, queues[1])
+
+			queues, err = exec.QueueList(ctx, 2, 2)
+			require.NoError(t, err)
+
+			require.Len(t, queues, 1)
+			requireQueuesEqual(t, queue3, queues[0])
+		})
+
+		t.Run("QueuePause", func(t *testing.T) {
+			t.Parallel()
+
+			t.Run("ExistingQueue", func(t *testing.T) {
+				t.Parallel()
+
+				exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+				queue := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Name: ptrutil.Ptr("queue1")})
+				require.Nil(t, queue.PausedAt)
+
+				require.NoError(t, exec.QueuePause(ctx, queue.Name))
+
+				queueFetched, err := exec.QueueGet(ctx, queue.Name)
+				require.NoError(t, err)
+				require.NotNil(t, queueFetched.PausedAt)
+				require.WithinDuration(t, time.Now(), *(queueFetched.PausedAt), 500*time.Millisecond)
+			})
+
+			t.Run("NonExistentQueue", func(t *testing.T) {
+				t.Parallel()
+
+				exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+				err := exec.QueuePause(ctx, "queue1")
+				require.ErrorIs(t, err, rivertype.ErrNotFound)
+			})
+
+			t.Run("AllQueues", func(t *testing.T) {
+				t.Parallel()
+
+				exec, _ := setupExecutor(ctx, t, driver, beginTx)
+
+				queue1 := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Name: ptrutil.Ptr("queue1")})
+				require.Nil(t, queue1.PausedAt)
+				queue2 := testfactory.Queue(ctx, t, exec, &testfactory.QueueOpts{Name: ptrutil.Ptr("queue2")})
+				require.Nil(t, queue2.PausedAt)
+
+				require.NoError(t, exec.QueuePause(ctx, "*"))
```

Comment:
Could we use some kind of global constant for this, even if it's just for use inside River's internal code?

I was trying to look for how wildcard pause/resume worked, and found this very awkward to search for. It's very short for one, there's no constant so you can't "find references", and "*" is a regex meaningful character of course, so searching for it is a little extra painful as you need to put in the right escapes.

Reply:
Refactored all instances of this to use `riverdriver.AllQueuesString`, lmk if you have a better idea. It's used across the driver and the main `river` package so the options were limited without making it an official "public" API.

Reply:
Moved to `rivercommon` :v: